### PR TITLE
Add focal with glibc-2.23 localedata

### PR DESCRIPTION
**Pull request by mble**
This commit adds a focal base image with older locale data (from
glibc-2.23), in order to provide a smoother transition between OS
versions.

The process is relatively straightforward:
1. Download glibc-2.23 (the donor) and glibc-2.31 (the version bundled
with focal).
2. Copy the localedata from 2.23 to 2.31.
3. Patch ld-identification to handle the old localedata format from 2.23
without erroring out. This doesn't have a material impact as very few
programs make use of `LC_IDENTIFICATION`.
4. Build our custom glibc-2.31 and install it into /opt/.
5. Patch locale-gen (from the locales package) to force installation of
locales into /usr/lib/locale.
6. Set PATH to include our custom build binaries, notably localedef.
7. Copy the SUPPORTED locales file (from a glibc-2.23 system).
8. Run locale-def and reconfigure the locales package to install all the
locales.

The end result is a glibc-2.31-based image with 2.23 localedata.

Ref: GUS-W-8548492

### CODE REVIEW

File: dod-locales-base-focal-2.23-locales/scripts/locale-gen-patch.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+                input=$USER_LOCALES/$input
+             fi
+        fi
+-       localedef $no_archive -i $input -c -f $charset $locale_alias $locale || :; \
++       localedef $no_archive -i $input -c -f $charset $locale_alias "/usr/lib/locale/$locale" || :; \
+        echo ' done'; \
+ done
+ echo "Generation complete."
+EOF
+git apply --ignore-space-change --ignore-whitespace patch-locale-gen.patch || exit
```
Comment (jdowning) [inline dod-locales-base-focal-2.23-locales/scripts/locale-gen-patch.sh:22]: Should this `exit 1` if the patch fails?

Comment (mble) [reply dod-locales-base-focal-2.23-locales/scripts/locale-gen-patch.sh:22]: @jdowning Good call – it should.

Comment (mble) [reply dod-locales-base-focal-2.23-locales/scripts/locale-gen-patch.sh]: @jdowning Updated, thanks.

File: dod-locales-base-focal-2.23-locales/scripts/localepatch.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+WORKING_DIR=${HOME}
+LOCALE_SOURCE_VERSION=$1
+TARGET_VERSION=$2
+LOCALE_ARCHIVE=/usr/lib/locale/locale-archive
+
+cd "${WORKING_DIR}" || exit && \
```
Comment (halorgium) [inline dod-locales-base-focal-2.23-locales/scripts/localepatch.sh:10]: i might be misunderstanding. 
shouldn't this script fail if the working dir is missing?
also, we have `set -e`, so should we eliminate some of the `|| exit 1` pieces?

**Review by halorgium — COMMENTED** [review]
i'm not super proficient here, but looks good.

File: dod-locales-base-focal-2.23-locales/scripts/localepatch.sh (hunk 2 above)
Comment (mble) [reply dod-locales-base-focal-2.23-locales/scripts/localepatch.sh:10]: @halorgium yeah, it should I write bash like I'm baby-proofing a house so everything is wrapped in multiple layers of guards. I can trim them.

**Review by halorgium — APPROVED** [review]
Baby proof is ok!

**Comment by mble** [comment]
Hah, CI. That's funny. This doesn't have CI.

**Pull request merged**